ballerina/validator_visitor.bal

```diff
@@ -345,6 +351,9 @@ class ValidatorVisitor {
                         self.updatePath(listItemInputValue.name);
                         self.validateListVariableValue(listItemValue, listItemInputValue, location, fieldName);
                         self.removePath();
+                    } else if listItemValue is () {
+                        string expectedTypeName = getOfType(listItemInputValue.'type).name.toString();
```

Comment:
Shall we remove the `toString()` here as well? We can use the `string?` as the type here.

Reply:
We may have to cast it then.

Reply:
Yes, we can use casting or typeguard, whatever suits here. Using `toString()` is not an ideal way to handle these cases. There are many complexities in using `toString()`.